### ISSUE TITLE
How to deal with the need for SP.WorkflowServices functionality?

### ISSUE THREAD

**Issue by politicsruinseverything** [pnp/pnpjs#1556]
Thank you for reporting an issue, suggesting an enhancement, or asking a question. We appreciate your feedback - to help the team understand your
needs please complete the below template to ensure we have the details to help. Thanks!

**Please check out the [Docs](https://pnp.github.io/pnpjs/) to see if your question is already addressed there. This will help us ensure our documentation covers the most frequent questions.**

### Category
- [ ] Enhancement
- [ ] Bug
- [X] Question
- [ ] Documentation gap/issue

### Version

Please specify what version of the library you are using: [ 2. 0.11       ]

Please specify what version(s) of SharePoint you are targeting: [ 2019        ]

*If you are not using the latest release, please update and see if the issue is resolved before submitting an issue.*

### Expected / Desired Behavior / Question
*If you are reporting an issue please describe the expected behavior. If you are suggesting an enhancement please
describe thoroughly the enhancement, how it can be achieved, and expected benefit. If you are asking a question, ask away!*

### Observed Behavior
*If you are reporting an issue please describe the behavior you expected to occur when performing the action. If you are making a
suggestion or asking a question delete this section.*

### Steps to Reproduce
*If you are reporting an issue please describe the steps to reproduce the bug in sufficient detail to allow testing. If you are making
a suggestion or asking a question delete this section.*

Hello. What is the current recommendation for folks building an Angular 2+ application who are using @pnp/sp and need to deal with SP.WorkflowServices? Is this on the roadmap or will Workflow calls never be a part of @pnp/sp? Thanks.

**Comment by patrick-rodgers** [comment]
Hi @politicsruinseverything - we don't have a feature roadmap as such for each piece of SP functionality, rather we wait until someone needs it. We haven't looked at adding methods for the workflow stuff but can certainly have a look as time permits. That is the limiting factor. But absolutely would include those methods.

From my end we will leave the issue open and investigate when we can, otherwise if someone has the time and wants to submit a PR would review and merge.

**Comment by politicsruinseverything** [comment]
Thanks, Patrick. I'm surprised this does not come up a lot but I suppose most folks deal with WF using PowerShell. In a large corporate environment where the developers do not have access to those tools, being able to request all instances of workflows for a list, sort them by their status, and kill and/or restart them is incredibly necessary. Obviously, behind the scenes there are a lot of limitations to what the native SP tools deliver over REST but with some creativity I've been able to accomplish everything I need using SP.Workflow.js but it is clunky. Having that be a part of pnp/sp would be pretty awesome.

My JS/Typescript skills are pretty good but I don't know if they are good enough to contribute to this project. I would be happy to spend some time contributing if I could get some friendly help from you top-tier devs. There just isn't quite enough documentation available for someone of my average skill level to put all of the pieces together...but I'm a fast learner :)

**Comment by patrick-rodgers** [comment]
@politicsruinseverything absolutely folks would be happy to help you out if you want to take a shot. We wouldn't be able to provide help in real time, but if you want to start a WIP PR and ask questions there we'd be very happy to have you contribute!!

As well on the docs, it is all about time. Always opportunities to improve, just not always time to do it. Thanks again for your interest!

**Comment by politicsruinseverything** [comment]
Until such time as I can figure out how to start a WIP PR (work in progress pull request, no?), here is what I did:
1. made a folder under my src/assets location in my Angular CLI project called scripts
2. saved a copy of the SharePoint scripts into this folder (microsoftajax.js, init.js, sp.init.js, sp.js, sp.runtime.js, and sp.workflowservice.js)
3. added <script tag references to all of these scripts to the bottom of my index.html page
4. in the Angular component where I want to access SP.WorkflowServices I declared a global variable just after the import statements (top) like this: 
`declare var SP:any;`
5. on the first line of my component's class definition I assign the global SP namespace to my variable like this:
 ```
 export class MyComponent implements OnInit {
    SP = window['SP'];
    ...
  }
```
6. in the method I want to call WorkflowServices I do this:
   ```
 const context = new SP.ClientContext(path); //path to SharePoint
    const web = context.get_web();
    const list = web.get_lists().getByTitle('My List Title Here');
    const svcMgr = new SP.WorkflowServices.WorkflowServicesManager(context, web);
```


I hope this helps someone while I am trying to learn the GitHub Git rules :)
